### PR TITLE
Cpu architecture optimization documentation

### PR DESCRIPTION
**Pull request by mratsim**
- [x] x86
- [x] ARM 

cc @arthurk @jakubgs

### CODE REVIEW

File: docs/cpu_features.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+git clone https://github.com/status-im/nim-blscurve
+cd nim-blscurve
+git submodule update --init
+nim c -r -d:danger --passC:"-D__BLST_PORTABLE__" --outdir:build benchmarks/bench_sha256.nim
+nim c -r -d:danger --outdir:build benchmarks/bench_sha256.nim
+```
+
+Due to tree hashing, hashing 32 bytes is the most important benchmark.
+
+**Without SSE3**
```
Comment (tersec) [inline docs/cpu_features.md]:
```suggestion
**Without SSSE3**
```

File: docs/cpu_features.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+```
+Backend: BLST, mode: 64-bit
+====================================================================================================================================
+
+SHA256 - 32B - BLST                                                         4524886.878 ops/s          221 ns/op          660 cycles
+SHA256 - 128B - BLST                                                        1776198.934 ops/s          563 ns/op         1689 cycles
+SHA256 - 5MB - BLST                                                              70.723 ops/s     14139678 ns/op     42419720 cycles
+```
+**With SSE3**
```
Comment (tersec) [inline docs/cpu_features.md]:
```suggestion
**With SSSE3**
```

File: docs/cpu_features.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+====================================================================================================================================
+
+SHA256 - 32B - BLST                                                         5376344.086 ops/s          186 ns/op          555 cycles
+SHA256 - 128B - BLST                                                        2183406.114 ops/s          458 ns/op         1376 cycles
+SHA256 - 5MB - BLST                                                              87.142 ops/s     11475557 ns/op     34427254 cycles
+```
+
+### BMI2 & ADX
+
+Intel: Broadwell, 2015\
```
Comment (tersec) [inline docs/cpu_features.md:60]: Trailing backslash?

**Review by jakubgs — APPROVED** [review]
This is great, thanks. My only tiny nitpick would be maybe removing some whitespace from the code blocks with benchmark results so they fit on standard GitHub rendeing of markdown.

File: docs/cpu_features.md (hunk 3 above)
Comment (mratsim) [reply docs/cpu_features.md:60]: When the markdown is rendering it forces a new line.

**Pull request merged**